src/gateway/routers/Aggregator.sol

```diff
@@ -46,18 +47,28 @@ contract Aggregator is Auth, IAggregator {
     /// @inheritdoc IAggregator
     function file(bytes32 what, address[] calldata routers_) external auth {
         if (what == "routers") {
-            require(routers_.length <= MAX_ROUTER_COUNT, "Aggregator/exceeds-max-router-count");
+            uint8 quorum_ = uint8(routers_.length);
+            require(quorum_ > 0, "Aggregator/empty-router-set");
+            require(quorum_ <= MAX_ROUTER_COUNT, "Aggregator/exceeds-max-router-count");
+
+            uint64 sessionId = 0;
+            if (routers.length > 0) {
+                // Increment session id if it is not the initial router setup and the quorum was decreased
+                Router memory prevRouter = validRouters[address(routers[0])];
```

Comment:
`routers[0]` returns an address so I don't think you have to explicitly cast it to an address :S

Reply:
True, good catch